elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java

```diff
@@ -289,17 +289,26 @@ public boolean matchesId(String checkId) {
      * @param requestScope the request scope
      * @return a filtered collection of resources loaded from the datastore.
      */
-    public static Set<PersistentResource> loadRecords(Class<?> loadClass, RequestScope requestScope) {
+    public static Set<PersistentResource> loadRecords(Class<?> loadClass, RequestScope requestScope,
+                                                      Optional<FilterExpression> filterExpression) {
         DataStoreTransaction tx = requestScope.getTransaction();
 
         if (shouldSkipCollection(loadClass, ReadPermission.class, requestScope)) {
             return Collections.emptySet();
         }
 
         Iterable list;
-        Optional<FilterExpression> filterExpression = requestScope.getLoadFilterExpression(loadClass);
+        Optional<FilterExpression> filter = filterExpression;
+        if(filterExpression.isPresent()) {
+            if(requestScope.getLoadFilterExpression(loadClass).isPresent()) {
```

Comment:
getLoadFilterExpression is not a free operation.  Let's do it once and use the result in all these if/else checks.